### PR TITLE
Add a telemetry doc

### LINKED ISSUE #2700: document how to add telemetry
As a developer, I have a simple checklist that I can go through when I want to add some telemetry so that I can avoid missing things or having to redo them.

### PR DESCRIPTION
**Pull request by dmose**
"As a developer, I have a simple checklist that I can go through when I want to add some telemetry so that I can avoid missing things or having to redo them."

This is a straw-man checklist for how to add telemetry items to the system add-on.  I'm not convinced it's necessarily right at this point, and I'd love some feedback from @emtwo, @ncloudioj, @k88hudson, @tspurway, and anyone else who cares to give it.

Fixes #2700

### CODE REVIEW

File: docs/v2-system-addon/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+1. - [ ] File or update your existing client-side implementation bug so that it points to the "user story issue".
+1. - [ ] Talk to Marina (@emtwo) about the dashboard you're hoping to see from the data you're adding, and work with her to create a data model (or request her review on it after the fact).
+1. - [ ] Update `system-addons/test/schemas/pings.js` with a commented JOI schema of your changes, and add tests to system-addons/test/unit/TelemetryFeed.test.js to exercise the ping creation.
+1. - [ ] Update [data_events.md](data_events.md) with an example of the data in question
+1. - [ ] Update any fields that you've added, deleted, or changed in [data_dictionary.md](data_dictionary.md)
+1. - [ ] Get review from Nan (@ncloudioj) and (XXX or?) Marina (@emtwo) on the data schema and the documentation changes.
+1. - [ ] Get review from a [data steward]() of the documentation changes to make sure this data collection fits under the "share additional data" pref, which is opt-in in 53-release, and opt-out in 55-nightly (XXX waiting on info from @tspurway for the correct general framing here?).
+1. - [ ] Implement as usual...
```
Comment (ncloudioj) [inline docs/v2-system-addon/telemetry.md]: In practice, we usually do one more thing to verify that the telemetry change makes its way to the data pipeline and eventually shows up in the database in the stage. You can ask Nan (@ncloudioj) to do this once the telemetry change lands in the code base.

**Comment by dmose** [comment]
I'd love to get feedback from @jaredkerim on this as well.

**Comment by dmose** [comment]
Note that I'll need to make some changes based on Tim's info before it's actually ready to land, in addition to getting some review.

**Comment by dmose** [comment]
(More review, that is)

File: docs/v2-system-addon/telemetry.md (hunk 1 above)
Comment (dmose) [inline docs/v2-system-addon/telemetry.md]: Added below; thanks.

**Comment by dmose** [comment]
@fmarier, I'm particularly interested in your review on the "data review" section, but I'm happy to hear your thoughts on the rest of this too, if you're so inclined.

**Comment by emtwo** [comment]
Looks good!

**Comment by dmose** [comment]
I'm going to go ahead and land this.  @tspurway, I'm happy to still accept comments if you've got them, and I can always do another PR for them.

**Comment by tspurway** [comment]
LGTM, Dan!  Sorry for late response 😬

**Comment by as-pine-proxy** [comment]
[Treeherder: 57a72d01712494ebfaccf6a2797aa6e725073f3d](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=57a72d01712494ebfaccf6a2797aa6e725073f3d)

**Pull request merged**